sycl/test/regression/kernel_name_inside_sycl_namespace.cpp

```diff
@@ -15,8 +15,8 @@
 #include <CL/sycl.hpp>
 
 struct A {};
-namespace sycl {
 struct B {};
+namespace sycl {
```

Comment:
I don't understand this change...

Reply:
This test case is compiled with flag "-D" "__SYCL_DISABLE_NAMESPACE_INLINE__".
Without this patch, at the second compile triple = "spir64-unknown-linux-sycldevice", a header file is generated with
the declaration of  B under the SYCL_INLINE_NAMESPACE:
__SYCL_INLINE_NAMESPACE(cl) {
	...
template <> struct KernelInfo<::sycl::B> {
	 ...
  }
template <> struct KernelInfo<::A> {
  }
}

At the 3rd compile, this generated header file is included and every thing compiles nice.


With this patch, the header file generated has these declarations for A and B.
template <> struct KernelInfo<struct sycl::B> {
...
}
template <> struct KernelInfo<struct A> {
	 ...	 
}
With this the 3rd compile fails with:
In file included from <built-in>:1:
/tmp/kernel_name_inside_sycl_namespace-8b9eeb.h:44:44: error: no struct named 'B' in namespace 'cl::sycl'
template <> struct KernelInfo<struct sycl::B> {
                                     ~~~~~~^
1 error generated.

To fix this I can either make the change that I am proposing which generates this in the header file:
__SYCL_INLINE_NAMESPACE(cl) {
...			    
template <> struct KernelInfo<struct B> {
	...
  }
};
template <> struct KernelInfo<struct A> {
   ...
}
}
or compile with "-D" "__SYCL_ENABLE_NAMESPACE_INLINE__" .

Reply:
In fact the test exposes regression introduced by this patch.
The test code is valid and the compiler should not complain. I would like to block the change until the test passes without change in tests source code.